src/main/java/Duke.java

```diff
@@ -1,10 +1,213 @@
+     * @throws DukeDescriptionException If command requires a description and it not given.
+     * @throws BadDescriptionException If description provided does not match the format required by command.
+     * @throws BadDateException If date provided does not match format required by command.
+     */
+    public void readCommand(Command command) throws DukeDescriptionException,
+            BadDescriptionException, BadDateException {
+        switch (command) {
+        case LIST:
+            System.out.println("Here are all your tasks:");
+            taskList.printTask();
+            break;
+        case DONE:
+            try {
+                String strDone = sc.nextLine();
+                if (strDone.isEmpty()) {
+                    throw new DukeDescriptionException("Empty Description");
+                }
+                int done = Integer.parseInt(strDone.substring(1)); //there must be a space between command and input
+                if (done <= 0 || done > taskList.getSize()) {
+                    throw new BadDescriptionException("Description for done cannot be "
+                            + done);
+                }
+                taskList.getTask(done - 1).setDone(true);
+                System.out.println("Nice! I've marked this task as done: \n" +
+                        "  " + taskList.getTask(done - 1));
+                storage.writeListIntoFile(taskList.getList());
+            } catch (NumberFormatException e) {
+                throw new BadDescriptionException("Non-Integer");
+            }
+            break;
+        case TODO:
+            String todo = sc.nextLine();
+            if (todo.isEmpty()) {
+                throw new DukeDescriptionException("Empty Description");
+            }
+            Task taskToDo = new Todo(todo.substring(1));
+            taskList.addTask(taskToDo);
+            System.out.println("I've added this task: \n" +
+                    "  " + taskToDo + "\nNow you have " +
+                    taskList.getSize() + " tasks in the list." );
+            storage.writeListIntoFile(taskList.getList());
+            break;
+        case EVENT:
+            String event = sc.nextLine();
+            if (event.isEmpty()) {
+                throw new DukeDescriptionException("Empty Description");
+            }
+            int eventDate = event.indexOf("/");
+            if (eventDate == -1) {
+                throw new BadDescriptionException("Missing '/' in Description");
+            }
+            Task taskEvent = new Event(event.substring(1, eventDate),
+                    event.substring(eventDate + 4));
+            taskList.addTask(taskEvent);
+            calender.addDate(taskEvent);
+            System.out.println("I've added this task: \n" +
+                    "  " + taskEvent + "\nNow you have " +
+                    taskList.getSize() + " tasks in the list." );
+            storage.writeListIntoFile(taskList.getList());
+            break;
+        case DEADLINE:
```

Comment:
These case blocks seem rather long. Perhaps it would be better to use helper functions to make the code more readable. Moreover, creating more helper functions helps in code maintenance later on.